sdk/compiler/daml-extension/src/extension.ts

```diff
@@ -67,69 +68,76 @@ export async function activate(context: vscode.ExtensionContext) {
-          // Reload configuration from VSCode
-          const config = vscode.workspace.getConfiguration("daml");
-          const consent = await getTelemetryConsent(config, context);
-          // Recalculate server args
-          const serverArgs = getLanguageServerArgs(config, consent);
-          console.log(serverArgs);
-          // Assign them to the LanguageServer (in a way that VSCode likely doesn't like)
-          serverExecutable.args = serverArgs;
-          // Stop the Language server
-          stopKeepAliveWatchdog();
-          await damlLanguageClient.stop();
-          // Give it a moment
-          await new Promise(resolve => setTimeout(resolve, 1000));
-          // Start it back up again using the new args
-          damlLanguageClient.start();
-          startKeepAliveWatchdog();
-        }
-      },
+  async function shutdownLanguageServer() {
+    // Stop the Language server
+    stopKeepAliveWatchdog();
+    await damlLanguageClient.stop();
+    virtualResourceManager.dispose();
+    const index = context.subscriptions.indexOf(virtualResourceManager, 0);
+    if (index > -1) {
+      context.subscriptions.splice(index, 1);
+    }
+  }
+
+  async function setupLanguageServer(
+    config: vscode.WorkspaceConfiguration,
+    consent: boolean | undefined,
+  ) {
+    damlLanguageClient = createLanguageClient(config, consent);
+    damlLanguageClient.registerProposedFeatures();
+
+    virtualResourceManager = new VirtualResourceManager(
+      damlLanguageClient,
+      webviewFiles,
+      context,
     );
-  });
+    context.subscriptions.push(virtualResourceManager);
+
+    let _unused = damlLanguageClient.onReady().then(() => {
+      startKeepAliveWatchdog();
+      damlLanguageClient.onNotification(
+        DamlVirtualResourceDidChangeNotification.type,
+        params =>
+          virtualResourceManager.setContent(params.uri, params.contents),
+      );
+      damlLanguageClient.onNotification(
+        DamlVirtualResourceNoteNotification.type,
+        params => virtualResourceManager.setNote(params.uri, params.note),
+      );
+      damlLanguageClient.onNotification(
+        DamlVirtualResourceDidProgressNotification.type,
+        params =>
+          virtualResourceManager.setProgress(
+            params.uri,
+            params.millisecondsPassed,
+            params.startedAt,
+          ),
+      );
+    });
+
+    damlLanguageClient.start();
+  }
 
-  damlLanguageClient.start();
+  vscode.workspace.onDidChangeConfiguration(
+    async (event: vscode.ConfigurationChangeEvent) => {
+      if (event.affectsConfiguration("daml")) {
+        await shutdownLanguageServer();
+        await new Promise(resolve => setTimeout(resolve, 1000));
+        const config = vscode.workspace.getConfiguration("daml");
+        const consent = await getTelemetryConsent(config, context);
+        setupLanguageServer(config, consent);
+      }
+    },
+  );
+
+  setupLanguageServer(config, consent);
```

Comment:
Should we await `getTelemetryConsent` before this first launch as well? In fact, what initializes the `consent` variable on this line?

Reply:
Line 54 gets the consent variable, and is already awaited. I'll move the lines closer though to match the changed configuration handler